core/kazoo_endpoint/src/kz_privacy.erl

```diff
@@ -112,7 +117,36 @@ is_anonymous(JObj) ->
     IsCallerNumberZero
         orelse IsPrivacyName
         orelse IsPrivacyNumber
-        orelse HasPrivacyFlags.
+        orelse HasPrivacyFlags
+        orelse check_anonymous_cid_number(JObj)
+        orelse check_anonymous_cid_name(JObj).
+
+-spec check_anonymous_cid_number(kz_json:object()) -> boolean().
+check_anonymous_cid_number(JObj) ->
+    case kapps_config:get_is_true(?PRIVACY_CAT, <<"check_additional_anonymous_cid_numbers">>, 'false') of
+        'true' -> is_anonymous_cid_number(JObj);
+        'false' -> 'false'
+    end.
+
+-spec check_anonymous_cid_name(kz_json:object()) -> boolean().
+check_anonymous_cid_name(JObj) ->
+    case kapps_config:get_is_true(?PRIVACY_CAT, <<"check_additional_anonymous_cid_names">>, 'false') of
+        'true' -> is_anonymous_cid_name(JObj);
+        'false' -> 'false'
+    end.
+
+-spec is_anonymous_cid_number(kz_json:object()) -> boolean().
+is_anonymous_cid_number(JObj) ->
+    lists:member(kz_json:get_value(<<"Caller-ID-Number">>, JObj)
+                ,kapps_config:get_ne_binaries(?PRIVACY_CAT, ?KEY_ANONS, ?DEFAULT_ANON_CIDS)
+                ).
+
+-spec is_anonymous_cid_name(kz_json:object()) -> boolean().
+is_anonymous_cid_name(JObj) ->
+    lists:member(kz_json:get_value(<<"Caller-ID-Name">>, JObj)
```

Comment:
Let's use typed-accessors where possible: `kz_json:get_ne_binary_value/2` in these cases